File: Problems/Leetcode/Java/LC412FizzBuzz.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class LC412FizzBuzz {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        scan.close();

        List<String> arr = new ArrayList<>();

        for (int i = 1; i < n + 1; i++) {
            arr.add(String.valueOf(i));
        }

        for (int i = 2; i < n + 1; i += 3) {
            if (i >= arr.size())
                break;
            arr.set(i, "Fizz");
        }

        for (int i = 4; i < n + 1; i += 5) {
            if (i >= arr.size())
                break;
            arr.set(i, "Buzz");
        }

        for (int i = 14; i < n + 1; i += 15) {
            if (i >= arr.size())
                break;
            arr.set(i, "FizzBuzz");
        }

        for (String i : arr) {
            System.out.println(i);
        }
    }
}

File: Problems/Leetcode/Python/LC412FizzBuzz.py
def fizzBuzz(n):
    arr = [str(i) for i in range(1, n+1)]
    
    for i in range(2, n, 3):
        arr[i] = "Fizz"
        
    for i in range(4, n, 5):
        arr[i] = "Buzz"
        
    for i in range(14, n, 15):
        arr[i] = "FizzBuzz"
    
    return arr


n = int(input())
print(fizzBuzz(n))
